web/src/app/schedules/ScheduleRuleCreateDialog.js

```diff
@@ -1,79 +1,96 @@
 import { fieldErrors, nonFieldErrors } from '../util/errutil'
 import gql from 'graphql-tag'
 import { startCase } from 'lodash-es'
+import { DateTime } from 'luxon'
+import { isoToClockTime } from './util'
 
 const mutation = gql`
   mutation($input: ScheduleTargetInput!) {
     updateScheduleTarget(input: $input)
   }
 `
-
-export default class ScheduleRuleCreateDialog extends React.PureComponent {
-  static propTypes = {
-    scheduleID: p.string.isRequired,
-    targetType: p.oneOf(['rotation', 'user']).isRequired,
-    onClose: p.func,
+const query = gql`
+  query($id: ID!) {
+    schedule(id: $id) {
+      id
+      timeZone
+    }
   }
+`
+
+export default function ScheduleRuleCreateDialog(props) {
+  const { scheduleID, targetType, onClose } = props
+  const [value, setValue] = useState({
+    targetID: '',
+    rules: [
+      {
+        start: DateTime.local()
+          .startOf('day')
+          .toUTC()
+          .toISO(),
+        end: DateTime.local()
+          .plus({ day: 1 })
+          .startOf('day')
+          .toUTC()
+          .toISO(),
+        weekdayFilter: [true, true, true, true, true, true, true],
+      },
+    ],
+  })
 
-  state = {
-    value: {
-      targetID: '',
-      rules: [
-        {
-          start: '00:00',
-          end: '00:00',
-          weekdayFilter: [true, true, true, true, true, true, true],
+  const { data, ...queryStatus } = useQuery(query, {
+    variables: { id: scheduleID },
+  })
+  const [mutate, mutationStatus] = useMutation(mutation, {
+    onCompleted: onClose,
+    variables: {
+      input: {
+        target: {
+          type: targetType,
+          id: value.targetID,
         },
-      ],
-    },
-  }
+        scheduleID,
 
-  render() {
-    return (
-      <Mutation mutation={mutation} onCompleted={this.props.onClose}>
-        {this.renderDialog}
-      </Mutation>
-    )
-  }
+        rules: value.rules.map(r => ({
+          ...r,
+          start: isoToClockTime(r.start, data.schedule.timeZone),
+          end: isoToClockTime(r.end, data.schedule.timeZone),
+        })),
+      },
+    },
+  })
 
-  renderDialog = (commit, status) => {
-    return (
-      <FormDialog
-        onClose={this.props.onClose}
-        title={`Add ${startCase(this.props.targetType)} to Schedule`}
-        errors={nonFieldErrors(status.error)}
-        maxWidth='md'
-        onSubmit={() => {
-          commit({
-            variables: {
-              input: {
-                target: {
-                  type: this.props.targetType,
-                  id: this.state.value.targetID,
-                },
-                scheduleID: this.props.scheduleID,
+  return (
+    <FormDialog
+      onClose={onClose}
+      title={`Add ${startCase(targetType)} to Schedule`}
+      errors={nonFieldErrors(mutationStatus.error)}
+      maxWidth='md'
+      loading={queryStatus.loading || mutationStatus.loading}
+      onSubmit={() => {
+        mutate()
+      }}
```

Comment:
We can't do that as `onSubmit` is an event handler -- it will pass in arguments not meant for `mutate` and we don't want it to interpret the return value of `mutate`